lib/browser/growl.js

```diff
@@ -63,23 +63,29 @@ exports.notify = function(runner) {
  */
 function isPermitted() {
   var permitted = {
-    granted: Promise.resolve(true),
-    denied: Promise.resolve(false),
-    default: (function ask() {
-      Notification.requestPermission().then(function(permission) {
-        return Promise.resolve(permission === 'granted');
+    granted: function allow() {
+      return Promise.resolve(true);
+    },
+    denied: function deny() {
+      return Promise.resolve(false);
+    },
+    default: function ask() {
+      return new Promise(function(resolve, reject) {
+        Notification.requestPermission().then(function(permission) {
+          return resolve(permission === 'granted');
+        });
       });
-    })()
+    }
   };
 
-  return permitted[Notification.permission];
+  return permitted[Notification.permission]();
 }
 
 /**
  * A promise determining if notification can proceed.
  *
  * @promise Growl~CanNotifyPromise
- * @fulfill {boolean} Notification allowed (<code>true</code>).
+ * @fulfill {undefined} Notification allowed.
```

Comment:
1.  Does jsdoc do anything with these?  Last I saw it wasn't implemented.  Would it be problematic to add a `@returns Promise<void> Notification allowed` if the API docs don't catch this?
1.  Please mark new methods `@private` if they are not intended to be consumed externally